### ISSUE TITLE
Shader effects are multipled on screenshots

### ISSUE THREAD

**Issue by Xenfo** [hyprwm/Hyprland#2946]
### Hyprland Version

Hyprland, built from branch main at commit 126792584fa4eb25f6c8e3b3eb6bfb00b58a787a dirty (renderer: don't add workspace offset to pinned). Tag: v0.28.0-28-g12679258  flags: (if any)

### Bug or Regression?

Bug

### Description

When taking screenshots, shader effects are multipled.

### How to reproduce

1. Enable the shader @ https://github.com/Xenfo/dotfiles/blob/main/hypr/shaders/blue-light-filter.glsl
2. Take a screenshot (I used Flameshot)
3. The image should look orange while the monitor looks normal

### Crash reports, logs, images, videos

![image](https://github.com/hyprwm/Hyprland/assets/46873232/59f15a88-5fd3-431d-9317-8f1811552c2e)

**Comment by MahouShoujoMivutilde** [comment]
That's because shaders are captured on screenshots too. It's not like redshift on xorg (which is more like another layer that doesn't affect image capture in any way).

So you really are applying a shader twice - first rendered in image that you captured, and second by viewing it with shader active.

Consider disabling a shader when taking a screenshot.

```glsl
// blank shader

precision mediump float;
varying vec2 v_texcoord;
uniform sampler2D tex;

void main() {
    vec4 pixColor = texture2D(tex, v_texcoord);
    gl_FragColor = pixColor;
}
```

```bash

blank=~/.config/hypr/blank.glsl

shader_changed=false

if [[ $XDG_CURRENT_DESKTOP == "Hyprland" ]]; then
    current="$(hyprctl getoption decoration:screen_shader -j | gojq -r '.str')"

    if [[ "$current" =~ (blank|EMPTY) ]] || [[ "$current" == "" ]] ||
        [[ "$current" == "$blank" ]];
    then
        echo 'shader: nothing needs to be done'
    else
        hyprctl keyword decoration:screen_shader "$blank"
        shader_changed=true
        echo "shader: set $blank"
    fi
fi

# actually take a screenshot 
# ...

# restore if needed
if $shader_changed; then
    hyprctl keyword decoration:screen_shader "$current"
fi

```

It's annoying, but that's a hacky workarounds for you instead of a proper solution (GAMMA_LUT implemented by Nvidia, which... by the looks of how it's going - will probably happen in next 2-3 years. Maybe, not promises are made. lmao).

**Comment by Xenfo** [comment]
![image](https://github.com/hyprwm/Hyprland/assets/46873232/075f063f-d2d1-46e5-bdb3-07970a75ff04)

So that does fix it, would there be any way to still have it apply the shader once? Would I have to compile a custom version of Flameshot that changes it right before it takes the final capture?

**Comment by MahouShoujoMivutilde** [comment]
Yes, but the others will still see just _slightly_ orange image.

Ideally, the capture would happen before the custom shaders (or at least it'll be configurable), but I am unaware how (not)difficult it will be to implement on hyprland side.

**Comment by Xenfo** [comment]
Alright, one more idea. Instead of having it flash blank (it really hurts my eyes, im just sensitive ig lol), would it be possible to invert shaders post screenshot? Since this is just a simple blue light filter.

**Comment by MahouShoujoMivutilde** [comment]
It most likely is possible, but I don't know how.

**Comment by Xenfo** [comment]
I wrote a little [something](https://github.com/Xenfo/imgmath) to reverse the effect for my version of the shader, so apart from screensharing, the issue is mostly fine for me.

**Comment by MahouShoujoMivutilde** [comment]
Nice!
